### ISSUE TITLE
Range Month Selector, Min-Date not considering year

### ISSUE THREAD

**Issue by umang1995** [Vuepic/vue-datepicker#333]
**Describe the bug**
When selecting min date for month selector, its only considering month for validation.

**To Reproduce**
Initialise the Datepicker with min-date as 2022-06-01,
As Expected, Months less than June will be disabled.
Now Select Month 2023-01 to 2023-02
**Expected behavior**
Datepicker should be validated.
**Actual behaviour**
Datepicker not validating
**Screenshots**
![image](https://user-images.githubusercontent.com/17137778/228639874-8409a5f0-d6b2-4682-8bf8-bc4b48770506.png)

**Desktop & mobile (please complete the following information):**
 - Browser [e.g. chrome, safari] - Chrome
 - Library version 4.2.3

**Comment by Jasenkoo** [comment]
@umang1995 Are you sure you are on the last version, there was an issue regarding this validation that is fixed in `4.2.3`. As you can see on this [demo](https://stackblitz.com/edit/vuepic-vue-datepicker-nhggc6?file=src%2Fcomponents%2FPlayground.vue), it looks ok.

**Comment by umang1995** [comment]
@Jasenkoo  Hey Yeah, I am on the latest version,  I just checked, so the issue is a little tricky, I have updated the [demo](https://stackblitz.com/edit/vuepic-vue-datepicker-bx8gom?file=src%2Fcomponents%2FPlayground.vue). 
Let me rephrase the issue - 

So Lets say any date between June 2022 to Feb 2023 should be acceptable,
so i keep:
-  min-date as new Date(2022, 5, 1)
- and max-date as new Date(2023,1,28).

As Expected , Mar23 and onwards are disabled, and May 2022 and before are disabled.

But If I am setting the v-model as 
[
  { month: 2, year: 2023 },
  { month: 2, year: 2023 },
]

This should be accceptable.

Also If I select say - 
Jan 2023 to Feb 2023 its also acceptable, 

I cannot change max-date to new Date(2023,2,1) since that enables march.

Please let me know if you understood, Or require any further clarifications
